### PR TITLE
Sign windows dll files.

### LINKED ISSUE #12751: Wazuh shared libraries are not signed.
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.2.5-rev | Wazuh component | Agent | Packages/Sources | Windows |

# Description
This issue aims to sign the dynamic libraries, so that they can have a better evaluation, under different malware scanners, from third party vendors.

Many times calls to operating system libraries, or some intrusion to obtain information relevant to the need of the product may seem strange for a third party software, when this software tries to help security.

### Actual
image

### Expected
Dll files signed.

### DoD
- [ ] Sign DLL files.
- [ ] Manual test with Get-AuthenticodeSignature

### PR DESCRIPTION
**Pull request by Dwordcito**
|Related issue|
|---|
|Closes #12751  |

## Description

This issue aims to sign the dynamic libraries, so that they can have a better evaluation, under different malware scanners, from third party vendors.

Many times calls to operating system libraries, or some intrusion to obtain information relevant to the need of the product may seem strange for a third party software, when this software tries to help security.

### Actual
image

### Expected
Dll files signed.

## Tests

- Compilation without warnings in every supported platform
  - [ ] Linux
  - [x] Windows
  - [ ] MAC OS X
- [ ] Source installation
- [x] Package installation
- [ ] Source upgrade
- [ ] Package upgrade
- [ ] Review logs syntax and correct language
- [ ] QA templates contemplate the added capabilities

- Memory tests for Linux
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Valgrind (memcheck and descriptor leaks check)
  - [ ] Dr. Memory
  - [ ] AddressSanitizer
- Memory tests for Windows
  - [ ] Scan-build report
  - [ ] Coverity
  - [ ] Dr. Memory
- Memory tests for macOS
  - [ ] Scan-build report
  - [ ] Leaks
  - [ ] AddressSanitizer

- [ ] Retrocompatibility with older Wazuh versions
- [ ] Working on cluster environments
- [ ] Configuration on demand reports new parameters
- [ ] The data flow works as expected (agent-manager-api-app)
- [ ] Added unit tests (for new features)
- [ ] Stress test for affected components

- Decoder/Rule tests
  - [ ] Added unit testing files ".ini"
  - [ ] runtests.py executed without errors

### CODE REVIEW

File: src/win32/wazuh-installer-sign-exes.bat [hunk 1]
```diff
@@ -2,8 +2,12 @@ SETLOCAL
 SET PATH=%PATH%;C:\Program Files\Microsoft SDKs\Windows\v7.0\Bin
 SET PATH=%PATH%;C:\Program Files (x86)\WiX Toolset v3.11\bin
 
-REM Fix all .exe files
+REM Fix all .exe and .dll files
 signtool.exe sign /a /tr http://rfc3161timestamp.globalsign.com/advanced /td SHA256 "*.exe"
+signtool.exe sign /a /tr http://rfc3161timestamp.globalsign.com/advanced /td SHA256 "..\*.dll"
+signtool.exe sign /a /tr http://rfc3161timestamp.globalsign.com/advanced /td SHA256 "..\data_provider\build\bin\sysinfo.dll"
+signtool.exe sign /a /tr http://rfc3161timestamp.globalsign.com/advanced /td SHA256 "..\shared_modules\dbsync\buildi\bin\dbsync.dll"
```
Comment (jnasselle) [inline src/win32/wazuh-installer-sign-exes.bat:9]:
```suggestion
signtool.exe sign /a /tr http://rfc3161timestamp.globalsign.com/advanced /td SHA256 "..\shared_modules\dbsync\build\bin\dbsync.dll"

```

**Comment by Dwordcito** [comment]
image
image

**Comment by Dwordcito** [comment]
DBSYNC
image

**Comment by Dwordcito** [comment]
LIBWAZUHEXT
image

**Comment by Dwordcito** [comment]
LIBWAZUHSHARED
image

**Comment by Dwordcito** [comment]
image

**Comment by Dwordcito** [comment]
SYSCOLLECTOR
image

**Comment by Dwordcito** [comment]
SYSINFO
image

**Review by alberpilot — APPROVED** [review]
LGTM

**Review by jnasselle — APPROVED** [review]

**Review by vikman90 — APPROVED** [review]

**Pull request merged**